src/tracer.test.js

```diff
@@ -52,53 +56,137 @@ describe('tracer', () => {
+  });
+
+  test('startTrace - timeout timer - simple flow', async done => {
+    const timeout = 1000;
+    const testBuffer = 50;
+
+    utils.setTimeoutTimerEnabled();
+    new EnvironmentBuilder().awsEnvironment().applyEnv();
+    const handlerInputs = new HandlerInputesBuilder()
+      .withTimeout(timeout)
+      .build();
+    TracerGlobals.setHandlerInputs(handlerInputs);
+
+    await tracer.startTrace();
+    SpansContainer.addSpan({ id: 'SomeRandomHttpSpan' });
+
+    setTimeout(() => {
+      const requests = HttpsRequestsForTesting.getRequests();
+      expect(requests.length).toEqual(2);
+      done();
+    }, timeout + testBuffer);
+  });
 
-    expect(spies.sendSingleSpan).toHaveBeenCalledTimes(0);
-    expect(spies.SpansContainer.addSpan).toHaveBeenCalledTimes(0);
+  test('startTrace - timeout timer - called twice', async done => {
+    const timeout = 1000;
+    const testBuffer = 50;
+
+    utils.setTimeoutTimerEnabled();
+    new EnvironmentBuilder().awsEnvironment().applyEnv();
+    const handlerInputs = new HandlerInputesBuilder()
+      .withTimeout(timeout)
+      .build();
+    TracerGlobals.setHandlerInputs(handlerInputs);
+
+    await tracer.startTrace();
+    await tracer.startTrace();
+    SpansContainer.addSpan({ id: 'SomeRandomHttpSpan' });
+
+    setTimeout(() => {
+      const requests = HttpsRequestsForTesting.getRequests();
+      //Expect 3 - 2 start spans and 1 from the timer
+      expect(requests.length).toEqual(3);
+      done();
+    }, timeout + testBuffer);
+  });
+
+  test('startTrace - timeout timer - to short timeout (timer not effects)', async done => {
+    const timeout = 10;
+    const testBuffer = 50;
+
+    utils.setTimeoutTimerEnabled();
+    new EnvironmentBuilder().awsEnvironment().applyEnv();
+    const handlerInputs = new HandlerInputesBuilder()
+      .withTimeout(timeout)
+      .build();
+    TracerGlobals.setHandlerInputs(handlerInputs);
+
+    await tracer.startTrace();
+    SpansContainer.addSpan({ id: 'SomeRandomHttpSpan' });
+
+    setTimeout(() => {
+      const requests = HttpsRequestsForTesting.getRequests();
+      //Expect 1 for start span
+      expect(requests.length).toEqual(1);
+      done();
+    }, timeout + testBuffer);
+  });
+
+  test('startTrace - timeout timer - SEND_ONLY_ON_ERROR - not sending spans', async done => {
+    const timeout = 1000;
+    const testBuffer = 50;
+
+    utils.setTimeoutTimerEnabled();
+    utils.setSendOnlyIfErrors();
+    new EnvironmentBuilder().awsEnvironment().applyEnv();
+    const handlerInputs = new HandlerInputesBuilder()
+      .withTimeout(timeout)
+      .build();
+    TracerGlobals.setHandlerInputs(handlerInputs);
+
+    await tracer.startTrace();
+    await tracer.startTrace();
+    SpansContainer.addSpan({ id: 'SomeRandomHttpSpan' });
+
+    setTimeout(() => {
+      const requests = HttpsRequestsForTesting.getRequests();
+      expect(requests.length).toEqual(0);
+      done();
+    }, timeout + testBuffer);
```

Comment:
Same as the test above.

Reply:
its 10ms, I don't want the test to be flaky